docs/handlers.md

```diff
@@ -0,0 +1,118 @@
+synchronous code was paused at the `await` on line `3` (well, more precisely, somewhere
+_within_ the `em.persist()` call). Only afterwards could the asynchronous code in the
+handler's `log()` be resumed - but the `User` object contained in the log entry the handler
+is still processing has already been mutated, and so the handler will incorrectly log
+something roughly equivalent to `Resolved user: { name: "${request.newName}" }`.
+
+So how do you avoid this? Well, if your handler serialises the entry in some way, e.g. to
+send it over the wire to an external service, then the only thing you need to do is make
+sure this serialisation takes place during the synchronous part of code execution, that is,
+before the code stops to `await` something. But also note that when you `await someFunctionCall()`,
+then the synchronous code _within_ `someFunctionCall()` also gets executed synchronously
+at the point the function is called, and only when an `await` is encountered is execution
+paused and a Promise is returned to be awaited on the previous level. Thus, for example,
+the [Elastic handler] doesn't suffer from this issue, even though it doesn't explicitly
+do anything to avoid it - because its `log()` method synchronously delegates to the Elastic
+SDK's `index()` method, which, also synchronously, serialises the entire request payload
+using `JSON.stringify()`. So there is no chance of user-land code possibly modifying the
+entry data between the data entering Debugr internals and being serialised to be sent
+to Elastic.
+
+If your code _doesn't_ serialise the entry, your other option is to use the `clone()`
+helper exported from `@debugr/core`. This function uses the V8 `serialize()` and `deserialize()`
+functions, which are faster than `JSON.stringify()` and `JSON.parse()` and will also
+be able to handle some native objects which cannot be represented with JSON (e.g. `Date`s).
+
+## Output formatting
+
+A handler usually either forwards the data elsewhere for further processing in some sort of
+machine-readable format, or directly generates some kind of human-readable output. In the latter
+case handlers can use Formatter plugins to generate rich semantic output from specialised
+log entries. A Formatter plugin always translates exactly one specialised entry type into
+format expected by exactly one Handler plugin. Packages providing a Handler plugin may optionally
+also provide default Formatters for that specific Handler plugin and there is a utility in Debugr
+core which simplifies some of the initial setup.
+
+(todo: more info about how handlers _use_ formatters)
+
+## Tasks
+
+Handlers may or may not care about Debugr tasks. Handlers which do care need to implement
+the `runTask()` method; this method will be called from within `Logger.runTask()`. When calling
+`Logger.runTask()`, Debugr constructs a "Russian doll" out of all the handlers which implement
```

Comment:
Ukrainian? :D

Reply:
umm... taky jsem nad tim premyslel, asi to zmenim na nejakej uplne jinej popis